File: lib/src/model/HeadObjectResult.java
package com.alibaba.sdk.android.oss.model;

/**
 * Created by zhouzhuo on 11/24/15.
 */
 class HeadObjectResult extends OSSResult {

    // object metadata
     ObjectMetadata metadata = new ObjectMetadata();

     ObjectMetadata getMetadata() {
        return metadata;
    }

     void setMetadata(ObjectMetadata metadata) {
        this.metadata = metadata;
    }

    @Override
     String toString() {
        String desc = String.format("HeadObjectResult<%s>:\n metadata:%s", super.toString(), metadata.toString());
        return desc;
    }
}

File: lib/src/model/head_object_result.dart
 import 'object_metadata.dart';

import 'oss_result.dart';

class HeadObjectResult extends OSSResult {

    // object metadata
     ObjectMetadata metadata = ObjectMetadata();

    @override
     String toString() {
        return "HeadObjectResult<${super}>:\n metadata:$metadata";
    }
}
